core/include/thread.h

```diff
@@ -454,11 +454,13 @@ static inline const char *thread_getname(kernel_pid_t pid)
  *
  * Only works if the thread was created with the flag THREAD_CREATE_STACKTEST.
  *
- * @param[in] stack the stack you want to measure. Try `thread_get_stackstart(thread_get_active())`
+ * @param[in] stack     the stack you want to measure. Try
+ *                      `thread_get_stackstart(thread_get_active())`
+ * @param[in] size      size of @p stack in bytes
  *
- * @return          the amount of unused space of the thread's stack
+ * @return              the amount of unused space of the thread's stack
  */
-uintptr_t thread_measure_stack_free(const char *stack);
+uintptr_t thread_measure_stack_free(const char *stack, size_t size);
```

Comment:
Why not just pass a thread struct pointer here instead? That'd make it a typed pointer that's easier to document and harder to mix up. (Either works for https://github.com/RIOT-OS/rust-riot-wrappers/pull/96 which is where I'm coming from)

Reply:
OK, thanks for the explanation. If this were a higher level API I'd suggest having an internal low-level tool and high-level functions for "of thread *thread", "of the ISR thread" and "of the idle pseudothread", but as this is mostly used internally, fine.

Reply:
Maybe I should rename it to `measure_stack_free()` and provide a `thread_measure_stack_free()` as static inline wrapper for convenience?

Reply:
Wonderful :) With the `rust-riot-wrappers` working with both variants, I reordered the commits to bump that first so that `git biset` is more fun :)